### ISSUE TITLE
Add missing C MPI datatypes to mpi_f08

### ISSUE THREAD

**Issue by jsquyres** [open-mpi/ompi#5801]
Per https://github.com/open-mpi/ompi/pull/5799#pullrequestreview-159873191, we are missing the C MPI datatypes in `mpi_f08` (e.g., MPI_INT, MPI_CHAR, MPI_OFFSET, MPI_COUNT, ...etc.).

This prevents us from building against the MPICH test suite.  Here's how I tried to build the MPICH test suite:

```sh
#!/bin/sh

# First, build and install Open MPI and ensure it is in your path.

git clone git@github.com:pmodels/mpich.git
cd mpich
git submodules init
git submodules update
./autogen.sh
./configure --prefix=$HOME/bogus CC=gcc CXX=g++ FC=gfortran
cd test

# Edit mpi/include/mpitestconf.h to reset anything necessary
# E.g., on my system, mpich decided that INTEGER16 was available
# but Open MPI did not build support for it.  So I had to comment
# out the definition of HAVE_MPI_INTEGER16

make
```

**Comment by jsquyres** [comment]
Removed v3.0.x and v3.1.x targets.

**Comment by gpaulsen** [comment]
https://github.com/open-mpi/ompi/pull/5811 Merged into v4.0.x.  Closing
